src/data-workspace/data-workspace.js

```diff
@@ -15,11 +15,15 @@ import {
 import styles from './data-workspace.module.css'
 import { EntryForm } from './entry-form.js'
 import { FinalFormWrapper } from './final-form-wrapper.js'
+import { useHasCommentContext } from './has-comment/has-comment-context.js'
 
 export const DataWorkspace = ({ selectionHasNoFormMessage }) => {
     const { data } = useMetadata()
     const [dataSetId] = useDataSetId()
-    const initialDataValuesFetch = useDataValueSet()
+    const { populateHasCommentContextForDataSetValues } = useHasCommentContext()
+    const initialDataValuesFetch = useDataValueSet((data) => {
+        populateHasCommentContextForDataSetValues(data.dataValues)
+    })
```

Comment:
The `useDataValueSet` hook now accepts a param which represents the `onSuccess` property of the `react-query` `useQuery`. I guess it could be good to make this more explicit, like so:
```suggestion
    const initialDataValuesFetch = useDataValueSet({
        onSuccess: (data) => {
            populateHasCommentContextForDataSetValues(data.dataValues)
        },
    })
```

But I'm not actually sure about that, as it might create the expectation that you could add lots of different options here.

In any case, this solution definitely wasn't my first choice to begin with. I would have much preferred to have implemented a `useEffect` hook here with a dependency to `dataValueSet`. However, since that object gets refreshed way too often that would have caused more performance issues.

Reply:
If we are to keep this I would prefer your second example. We could just pass in an `options`-param, and spread it to `useQuery`.